Sylvester type definitions

dalie opened the pull request.

tkqubo committed:
Update _.partial

tkqubo committed:
Replace method

tkqubo committed:
test: write some tests and fix bugs

tkqubo committed:
feat: add catch-alls

vvakame committed:
Merge pull request #6234 from cherrry/def/bounce-js

Add defs + tests for Bounce.js

vvakame committed:
Merge pull request #6295 from chrootsu/lodash-toArray

lodash: signatures of the method _.toArray have been changed

vvakame committed:
Merge pull request #6322 from vvakame/fix-6297

remove unused reference at hammerjs/hammerjs.d.ts

vvakame committed:
Merge pull request #6296 from giggio/docopt

Add docopt

gcastre committed:
Improve durandal.d.ts if you use Q instead of jQuery

Durandal supports injection/configuration to use a different library for Deferred/Promises (ref http://durandaljs.com/documentation/Q.html)
When using a different Deferred/Promise implementation, you might want to use a different Promise interface in the durandal.d.ts file.

Added some type annotations to ensure that durandal.d.ts won’t make compilation fail when the compiling with the noImplicitAny option set to true.

Note that I'm not the author of these changes but was asked to review it, and then publish it for the benefits of the community.

vvakame committed:
Merge pull request #6305 from cyrilschumacher/master

Update FileSaver: add the parameter: "disableAutoBOM".

vvakame committed:
Merge pull request #6309 from laco0416/fix-polymer-beforeregister

Update polymer.d.ts

vvakame committed:
Merge pull request #6310 from bdoss/typeahead-templates

Added additional template methods to typeahead.js

vvakame committed:
Merge pull request #6311 from jvilk/mocha-fixes

[mocha] Adding definition of mocha.throwError, with test to cover it.

vvakame committed:
Merge pull request #6312 from fongandrew/analytics-flush

Flush for analytics-node

vvakame committed:
Merge pull request #6314 from kuy/master

Fix param name

Pro committed:
Extended definition for roslib to match library functions

vvakame committed:
Merge pull request #6316 from chrootsu/lodash-isEqual

lodash: signatures of a method _.isEqual (and of an alias _.eq) have …

vvakame committed:
Merge pull request #6317 from chrootsu/lodash-findLastKey

lodash: signatures of the method _.findLastKey have been changed

vvakame committed:
Merge pull request #6320 from MarkBouwman/angular-ui-router-reload

angular-ui-router: transitionTo() and reload() return promises

vvakame committed:
Merge pull request #6281 from Kopleman/patch-1

Adding missed cache to IState

vvakame committed:
Merge pull request #6323 from ekosystem/master

Fix ui-grid.d.ts syntax

vvakame committed:
Merge pull request #6293 from soycode/master

update freedom.js pgp interface

AnthonyMacKinnon commented on sylvester/sylvester-test.ts:
```diff
@@ -0,0 +1,207 @@
+v = v.to3D();
+m = v.toDiagonalMatrix();
+v = v.toUnitVector();
+v = v.x(n);
```
Should we test the global `$V/M/L/P` as well?

dalie replied:
Done

AnthonyMacKinnon commented:
LGTM

AnthonyMacKinnon commented on sylvester/sylvester-test.ts:
```diff
@@ -0,0 +1,207 @@
+m = m.inverse();
+bool = m.isSameSizeAs(m);
+bool = m.isSingular();
+bool = m.isSquare();
```
Wondering if we should add spacing in some of the big blocks...

dalie replied:
How do I decide where to put the spacing?

StephaneAlie commented on sylvester/sylvester.d.ts:
```diff
@@ -0,0 +1,825 @@
+// Vector and Matrix mathematics modules for JavaScript
+// Copyright (c) 2007 James Coglan
+
+interface VectorStatic {
```
While doing the def for proj4 I realized we could hide those interface in the module, otherwise they show up in the global scope, can you give it a try by moving all interfaces inside the Sylvester module? I'm guessing the global function might have to prefix some typing with Sylvester but it should work.

dalie replied:
Done

barbarosh committed:
Update function scope.

Add missed function toBeDefined.

alexeagle committed:
Angular 2 typings are now distributed via NPM

alexeagle committed:
Merge pull request #6328 from angular/no_tsd

Angular 2 typings are now distributed via NPM

robmhoward committed:
Removed license since it is already covered by repo

jfahrenkrug committed:
Update XRegExp to 3.0.0

chrootsu committed:
lodash: existing wrappers renamed to implicit chaining wrappers

chrootsu committed:
lodash: explicit chaining wrapper added; signatures of methods _.chain and _.tap changed

Commit:
Update polymer.d.ts to ver.1.1.5

vineus committed:
key should not be an array

mauricedb committed:
Added RefluxJS typings

mauricedb committed:
Added test

Commit:
add uikit

mauricedb committed:
Added functionality

mauricedb committed:
Fixed test errors

Commit:
copy README from uikit repo for typedoc porposes and quick update to documentation comments

kennep committed:
followRedirect is documented to also accept a function

mprobst committed:
feat: closure-compiler type definitions.

chrootsu committed:
lodash: signatures of the method _.thru changed

chrootsu committed:
iniparser: definition of the module has been added

Commit:
Error fixes and optimizations

Use VoxImplant namespace and rename module to voximplant-websdk (like
npm module)

Commit:
added forgeted function signature in uikit

louislarry committed:
add function seed to faker.d.ts

chrootsu committed:
utils-merge: definition of the module has been added

conficient committed:
Merge remote-tracking branch 'refs/remotes/borisyankov/master'

conficient committed:
ko.plus - amended signature of .fail method of command

Fail callback can be more than just a text response, added new callback
signature to address this.

abreits committed:
amqplib: Shared declarations combined

Mek7 committed:
Merge remote-tracking branch 'refs/remotes/borisyankov/master' into onsortchanged

Mek7 committed:
Added missing on.sortChanged handler (generic)

tkqubo committed:
Make connect return correct type

Pro committed:
fixed constructor. Has optional values

Pro committed:
Merge remote-tracking branch 'upstream/master'

Pro committed:
New definitions for jquery.highlight.js

jonnystoten committed:
Add some extras to knex that we've needed

gsipos committed:
angular-material: definitions for IDateLocaleProvider

gsipos committed:
angular-material: IDateLocaleProvider fix typo

jonnystoten committed:
Add test code for additions

jonnystoten committed:
Fix tests!

axelcostaspena committed:
Rename simpleStorage files in order to match naming convention.
Rename namespace including author name to a more convenient one.

Commit:
Use actual generated d.ts from webcola project (project is written in typescript)

basarat committed:
Merge pull request #6146 from AmazingJaze/master

Update to WinJS 4.4

chrootsu committed:
lodash: signatures of the methods _.prototype.commit, _.prototype.concat and _.now changed

chrootsu committed:
finalhandler: definition of the module has been added

chrootsu committed:
finalhandler: tests have been added

abreits committed:
amqplib: better description for shared properties module

horiuchi committed:
Merge pull request #6337 from kennep/master

followRedirect is documented to also accept a function

horiuchi committed:
Merge pull request #6342 from chrootsu/iniparser

iniparser: definition of the module has been added

horiuchi committed:
Merge pull request #6343 from aylarov/master

Error fixes and optimizations

horiuchi committed:
Merge pull request #6347 from chrootsu/utils-merge

utils-merge: definition of the module has been added

tkqubo committed:
make connect stricter

alexeagle committed:
Merge pull request #6340 from mprobst/closure-compiler

feat: closure-compiler type definitions.

Commit:
server.select() returns either a single server or a list of servers, not void

Commit:
Updating the 8.2 version as well

Commit:
Add firstpos1 and firstpos2 properties to PNotify stack

Per PNotify documentation:
https://github.com/sciactive/pnotify#example-stacks

chrootsu committed:
flat: definition of the module has been added

chrootsu committed:
flat: tests have been added

chrootsu committed:
lodash: signatures of the methods _.prototype.plant and _.prototype.reverse changed

vvakame committed:
Merge pull request #6327 from barbarosh/patch-1

Update function scope.

vvakame committed:
Merge pull request #6290 from Hamaze/master

Added additional IOptions properties

Commit:
Updating ObjectID to match current spec

vvakame committed:
Merge pull request #6332 from laco0416/polymer-1.1.5

Update polymer.d.ts to ver.1.1.5

vvakame committed:
Merge pull request #6331 from jfahrenkrug/xregexp-3.0.0

Update XRegExp to 3.0.0

Commit:
generate is not supposed to be a static function

vvakame committed:
Merge pull request #6321 from vineus/master

Fix async.forEachFor signature

vvakame committed:
Merge pull request #6333 from chrootsu/lodash-ExplicitChaining

lodash: an explicit chaining wrapper added

vvakame committed:
Merge pull request #6335 from giovannicandido/master

add uikit

vvakame committed:
Merge pull request #6336 from mauricedb/Add_reflux

Add RefluxJS

vvakame committed:
remove node_redis/node_redis.d.ts

vvakame committed:
Merge pull request #6307 from tkqubo/revert-webpack-merge

Revert "Merge webpack-env.* into webpack.*"

vvakame committed:
Merge pull request #5798 from alvarorahul/master

Add bundles, skipDataMain and onNodeCreated

vvakame committed:
Merge pull request #6346 from louislarry/patch-1

add function seed to faker.d.ts

vvakame committed:
Merge pull request #6349 from conficient/master

ko.plus - fixed .fail method signature on command type

vvakame committed:
Merge pull request #6355 from Pro/master

Extended definition for roslib to match library functions

use-strict committed:
webpack - added missing chunkName parameter

vvakame committed:
Merge pull request #6357 from tkqubo/react-redux-connect

Make connect return correct type

vvakame committed:
Merge pull request #5845 from axelcostaspena/master

Add simpleStorage definitions

vvakame committed:
Merge pull request #6109 from bluong/momentLongDate

Moment long date

vvakame committed:
Merge pull request #6361 from chrootsu/finalhandler

finalhandler: definition of the module has been added

vvakame committed:
Merge pull request #6286 from abreits/master

amqplib: callback api definition and tests added

vvakame committed:
Merge pull request #6367 from samherrmann/add-missing-properties-to-pnotify

Add firstpos1 and firstpos2 properties to PNotify stack

vvakame committed:
Merge pull request #6368 from chrootsu/flat

flat: definition of the module has been added

vvakame committed:
Merge pull request #6371 from BrandonCKrueger/update-mongodb-object-class

Updating ObjectID to match current spec

vvakame committed:
Merge pull request #6352 from MatejQ/onsortchanged

Added missing generic on.sortChanged handler

vvakame committed:
Merge pull request #6359 from Rocketmakers/knex-additions

Add some extra things for knex

robmhoward committed:
Removed license from test code

stpettersens committed:
Create express-handlebars.d.ts

stpettersens committed:
Create express-handlebars-tests.ts

Commit:
Create initial connect typings.

Commit:
Add tests for connect typings

Commit:
Fix HandleFunction typing issue

Commit:
Fix typings issues

Commit:
Add typings for connect-livereload

Commit:
Add tests for connect-livereload

abbasmhd committed:
Update dhtmlxscheduler.d.ts

use generic to strongly define the return type for `getEvent`

Commit:
Fix gruntjs typings: Add grunt.config.merge method

basarat committed:
Merge pull request #6380 from SomaticIT/fix-gruntjs

Fix gruntjs typings: Add grunt.config.merge method

nchen63 committed:
Update webcola.d.ts

horiuchi committed:
Merge pull request #6377 from SomaticIT/connect-typings

Connect typings

maeg02 committed:
Select2 has two template functions

bgrieder committed:
Merge remote-tracking branch 'upstream/master'

bgrieder committed:
react-intl: fixes for Travis build

bgrieder committed:
react-intl: more minor fixes for Travis build

bgrieder committed:
fix to avoid implicit Any

bgrieder committed:
fix invalid reference to react

chrootsu committed:
lodash: tests of the methods _.prototype.toString and _.prototype.value (with aliases) changed

bencoveney committed:
Updated Application constructor to accept options

chrootsu committed:
lodash: signatures of the method _.ceil changed

chrootsu committed:
lodash: signatures of the method _.inRange changed

chrootsu committed:
lodash: signatures of the method _.round changed

stpettersens committed:
Update express-handlebars.d.ts

stpettersens committed:
Update express-handlebars-tests.ts

chrootsu committed:
lodash: signatures of the method _.capitalize changed

stpettersens committed:
Update express-handlebars.d.ts

stpettersens committed:
Update express-handlebars.d.ts

horiuchi committed:
Merge pull request #6388 from maeg02/master

Select2 has two template functions

bencoveney committed:
Backbone: Fixed History.Started being undefined

The History.started flag is set on the class rather than on the class instance (as it is set once globally). Attempts to read *Backbone.history.started* (via the global history instance) always return undefined. Instead the typing should be static member (*Backbone.History.started* is where the correct value is located). This can be seen at http://backbonejs.org/docs/backbone.html#section-203

bbrink68 committed:
Correct Misspelled "IHttpRequestTransformer".

Was IHttpResquestTransformer.

Commit:
Fix issue #6065

* Update the Runner.run function.
* Update the Server.new function.
* Add new Config interface.
* Add new ConfigFile interface.
* Rename ClientConfig to ClientOptions.

chrootsu committed:
lodash: signatures of the method _.endsWith changed

chrootsu committed:
lodash: signatures of the method _.escapeRegExp changed

chrootsu committed:
lodash: signatures of the method _.attempt changed

chrootsu committed:
lodash: signatures of the method _.constant changed

chrootsu committed:
lodash: signatures of the method _.matches changed

chrootsu committed:
lodash: signatures of the method _.method changed

tkrotoff committed:
Refactoring

tkrotoff committed:
Should be knex-tests.ts instead of knex-test.ts

tkrotoff committed:
Rename from *-test.ts to *-tests.ts

Commit:
openlayers. Add named external module.

Enables usage of the module in imports.

danmarshall committed:
initial version

danmarshall committed:
fix errors

danmarshall committed:
added strong typing

pepaar committed:
Add definition for fixed-data-table

pepaar committed:
fixed-data-table d.ts added

pepaar committed:
test fix

chrootsu committed:
lodash: signatures of the method _.pad changed

suvjunmd committed:
Fixed typo

pksorensen committed:
Removed requirejs.d.ts double definition of bundles.

danmana committed:
Added definitions for jQuery.qrcode. Closes #6292

chrootsu committed:
lodash: signatures of the method _.padRight changed

chrootsu committed:
lodash: signatures of the method _.repeat changed

chrootsu committed:
lodash: signatures of the method _.mixin changed

chrootsu committed:
lodash: signatures of the method _.startCase changed

chrootsu committed:
lodash: signatures of the method _.template changed

Commit:
Add electron-packager typings

Commit:
Add electron-builder typings

SamuelMarks committed:
Correct package name: observerjs => observe-js

See: https://github.com/Polymer/observe-js/blob/master/package.json

dekelev committed:
Added missing flag 'v8debug' to Global interface

'v8debug' is an optional flag that will be set in NodeJS globals upon using the 'debug-brk' command parameter.
We're using it to dynamically detect when app is running in debug-mode.

(When not on debug-mode, the 'v8debug' flag is undefined)

dekelev committed:
Merge pull request #1 from gigya/v8debug-patch

Added missing flag 'v8debug' to Global interface

chrootsu committed:
lodash: signatures of the method _.trimLeft changed

chrootsu committed:
lodash: signatures of the method _.trunc changed

dflor003 committed:
DanilF - Added typings for decorum library.

vvakame committed:
Merge pull request #6373 from use-strict/patch-1

webpack - added missing chunkName parameter

vvakame committed:
Merge pull request #6358 from gsipos/master

angular-material: definitions for IDateLocaleProvider

vvakame committed:
Merge pull request #6378 from SomaticIT/connect-livereload-typings

Add connect-livereload typings

vvakame committed:
Merge pull request #6379 from abbasmhd/master

Update dhtmlxscheduler.d.ts

vvakame committed:
Merge pull request #6360 from nchen63/patch-4

Use actual generated d.ts from webcola project

Pro committed:
Renamed jquery.highlight to jquery.highlight-bartaz

vvakame committed:
Merge pull request #6372 from vvakame/fix-redis

remove node_redis/node_redis.d.ts

vvakame committed:
Merge pull request #6163 from use-strict/material-ui-export-types

material-ui - Also export types in named modules

vvakame committed:
Merge pull request #6303 from bgrieder/master

Definitions for react-intl 1.2.0

vvakame committed:
Merge pull request #6384 from chrootsu/lodash-toString-value

lodash: tests of the methods _.prototype.toString and _.prototype.valu…

vvakame committed:
Merge pull request #6387 from bencoveney/patch-1

Marionette: Updated Application constructor to accept options

vvakame committed:
Merge pull request #6389 from chrootsu/lodash-ceil

lodash: signatures of the method _.ceil changed

vvakame committed:
Merge pull request #6390 from chrootsu/lodash-inRange

lodash: signatures of the method _.inRange changed

vvakame committed:
Merge pull request #6391 from chrootsu/lodash-round

lodash: signatures of the method _.round changed

vvakame committed:
Merge pull request #6392 from chrootsu/lodash-capitalize

lodash: signatures of the method _.capitalize changed

vvakame committed:
Merge pull request #6393 from stpettersens/master

Type definitions and tests for express-handlebars

vvakame committed:
Merge pull request #6173 from wcomartin/master

add angular-dialog-service typescript definitions

vvakame committed:
Merge pull request #6313 from OfficeDev/master

New definitions for office-js library

vvakame committed:
Merge pull request #6197 from moonpyk/master

Knockout KnockoutUtils cleanup.

vvakame committed:
Merge pull request #6398 from bencoveney/patch-2

Backbone: Fixed History.Started being undefined

vvakame committed:
Merge pull request #6402 from bbrink68/patch-1

Correct Misspelled "IHttpRequestTransformer".

vvakame committed:
Merge pull request #6404 from chrootsu/lodash-endsWith

lodash: signatures of the method _.endsWith changed

vvakame committed:
Merge pull request #6405 from chrootsu/lodash-escapeRegExp

lodash: signatures of the method _.escapeRegExp changed

vvakame committed:
Merge pull request #6406 from chrootsu/lodash-attempt

lodash: signatures of the method _.attempt changed

vvakame committed:
Merge pull request #6407 from chrootsu/lodash-constant

lodash: signatures of the method _.constant changed

vvakame committed:
Merge pull request #6409 from chrootsu/lodash-matches

lodash: signatures of the method _.matches changed

vvakame committed:
Merge pull request #6411 from chrootsu/lodash-method

lodash: signatures of the method _.method changed

vvakame committed:
Merge pull request #6412 from tkrotoff/karma-refactoring

karma.d.ts refactoring

vvakame committed:
Merge pull request #6413 from tkrotoff/knex-refactoring

Should be knex-tests.ts instead of knex-test.ts

vvakame committed:
Merge pull request #6415 from tkrotoff/tests.ts

Rename from *-test.ts to *-tests.ts

vvakame committed:
Merge pull request #6417 from aindlq/openlayers-node-export

openlayers. Add named external module.

vvakame committed:
Merge pull request #6419 from danmarshall/master

initial version of openjscad

vvakame committed:
Merge pull request #6420 from pepaar/master

Facebook's fixed-data-table d.ts

vvakame committed:
Merge pull request #6421 from chrootsu/lodash-pad

lodash: signatures of the method _.pad changed

vvakame committed:
Merge pull request #6422 from suvjunmd/typo

Fixed typo

vvakame committed:
Merge pull request #6423 from s093294/patch-4

Removed requirejs.d.ts double definition of bundles.

vvakame committed:
Merge pull request #6426 from chrootsu/lodash-padRight

lodash: signatures of the method _.padRight changed

vvakame committed:
Merge pull request #6427 from chrootsu/lodash-repeat

lodash: signatures of the method _.repeat changed

vvakame committed:
Merge pull request #6428 from chrootsu/lodash-mixin

lodash: signatures of the method _.mixin changed

vvakame committed:
Merge pull request #6429 from chrootsu/lodash-startCase

lodash: signatures of the method _.startCase changed

vvakame committed:
Merge pull request #6430 from chrootsu/lodash-template

lodash: signatures of the method _.template changed

vvakame committed:
Merge pull request #6440 from chrootsu/lodash-trimLeft

lodash: signatures of the method _.trimLeft changed

vvakame committed:
Merge pull request #6441 from chrootsu/lodash-trunc

lodash: signatures of the method _.trunc changed

vvakame committed:
Merge pull request #6374 from tkqubo/lodash-partial

Update lodash _.partial and _.partialRight

vvakame committed:
Merge pull request #6434 from SomaticIT/electron-builder-packager

Add typings for packages 'electron-packager' and 'electron-builder'

vvakame committed:
Merge pull request #6436 from SamuelMarks/patch-1

Correct package name: observerjs => observe-js

chrootsu committed:
lodash: signatures of the method _.add changed

chrootsu committed:
lodash: signatures of the method _.floor changed

chrootsu committed:
lodash: signatures of the method _.words changed

chrootsu committed:
lodash: signatures of the method _.camelCase changed

chrootsu committed:
lodash: signatures of the method _.random changed

Commit:
Removing bad dependency ref

It's breaking the NuGet auto build:  see: https://ci.appveyor.com/project/JasonJarrett/nugetautomation/build/1.0.2356

vvakame committed:
Merge pull request #6437 from gigya/master

Added missing flag 'v8debug' to Global interface

vvakame committed:
Merge pull request #6444 from dflor003/library/decorum

DanilF - Added typings for decorum library.

vvakame committed:
Merge pull request #6324 from gcastre/durandal_q

Improve durandal.d.ts if you use Q instead of jQuery

vvakame committed:
Merge pull request #6356 from Pro/upstream

New definitions for jquery.highlight.js

vvakame committed:
Merge pull request #6403 from eyalsk/master

Fix issue #6065

vvakame committed:
Merge pull request #6448 from chrootsu/lodash-add

lodash: signatures of the method _.add changed

vvakame committed:
Merge pull request #6449 from chrootsu/lodash-floor

lodash: signatures of the method _.floor changed

vvakame committed:
Merge pull request #6450 from chrootsu/lodash-words

lodash: signatures of the method _.words changed

vvakame committed:
Merge pull request #6451 from chrootsu/lodash-camelCase

lodash: signatures of the method _.camelCase changed

vvakame committed:
Merge pull request #6452 from chrootsu/lodash-random

lodash: signatures of the method _.random changed

vvakame committed:
Merge pull request #6424 from danmana/master

Added definitions for jQuery.qrcode. Closes #6292

pksorensen committed:
made option config argument optional

pksorensen committed:
the configs for option is all optional

pksorensen committed:
added async,prompt and appname

pksorensen committed:
added gruntfile property and interface

pksorensen committed:
added types for making build happy

pksorensen committed:
added return types on methods

pksorensen committed:
A few more type errors

danmarshall committed:
added strong typed options

pksorensen committed:
last type error

horiuchi committed:
Merge pull request #6457 from s093294/patch-5

Few updates to yeoman

horiuchi committed:
Merge pull request #6366 from BrandonCKrueger/hapi-select-fix

server.select() returns a server, not void

Beng89 committed:
Added definitions and test for node-array-ext.

basarat committed:
Merge pull request #6458 from danmarshall/master

added strong typed options

horiuchi committed:
Merge pull request #6463 from Beng89/master

Added definitions and test for node-array-ext.

dalie committed:
Merge remote-tracking branch 'refs/remotes/borisyankov/master' into sylvester
